### ISSUE TITLE
Cannot copy param 0 weights from layer 'score-fr'; shape mismatch

### ISSUE THREAD

**Issue by caffecoffee** [martinkersner/train-CRF-RNN#35]
> F0311 16:30:12.422097 24027 net.cpp:861] Cannot copy param 0 weights from layer 'score-fr'; shape mismatch.  Source param shape is 21 4096 1 1 (86016); target param shape is 4 4096 1 1 (16384). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.
> *** Check failure stack trace: ***

I follow the instructions step by step. Everything works well ,until I execute 'python solve.py 2>&1 | tee train.log'. The error is shown as above. 
I try some other ways to solve it. But I still can't run through it.

If I change the layer's name, another error will appears.

> I0311 17:18:26.957114 24700 upgrade_proto.cpp:628] Successfully upgraded file specified using deprecated V1LayerParameter
> scoreme2 is composed of zeros!
> scoreme4 is composed of zeros!
> Exiting.

**Comment by qincao1994** [comment]
I encounter the same problem like this.
score2-new is composed of zeros!
Exiting.

**Comment by yxliwhu** [comment]
The reason you get the error is the .caffemodel you used is not correct. You should train you own FCN-8 model with own data firstly. And then, replacing the 'fcn-8s-pascal.caffemodel' in solove.py file.

**Comment by qincao1994** [comment]
@yxliwhu Thank you very much for your reply. I will have a try.

**Comment by caffecoffee** [comment]
@yxliwhu Thank you for your proposal! Very helpful!

**Comment by manyaafonso** [comment]
I have the same problem. How does one train without the initial weights?

@caffecoffee and @qincao1994 did this actually work?

**Comment by gopi77** [comment]
Hi

Got the same issue
>>> Cannot copy param 0 weights from layer 'score-fr'; shape mismatch.  Source param shape is 21 4096 1 1 (86016); target param shape is 4 4096 1 1 (16384). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.

I understand this project is for training the CRF-RNN. But as per the above comment from yxliwhu, do we need to do some other training before this training? bit confusing, please clarify.

**Comment by ThienAnh** [comment]
Hi
I got the same issue. 
@caffecoffee , @qincao1994 , @manyaafonso and @gopi77 : did you fix it?

**Comment by manyaafonso** [comment]
@ThienAnh I gave up on this particular repository, but I managed to train crfasrnn with my own data following these steps:
https://github.com/bittnt/caffe/blob/22babdb4a70eb5829c52fccc417d4b2f916d2748/examples/segmentationcrfasrnn/TVG_CRFRNN.sh

I initialized with the weights for the VOC dataset and renamed a few layers as necessary.

**Comment by ThienAnh** [comment]
@manyaafonso  Thank for reply. I will try it.

**Comment by ThienAnh** [comment]
HI @manyaafonso . I had fix this error. But when i'm training with mode GPU, have new an error : 

I0929 10:38:58.933909  4800 solver.cpp:571] Iteration 1150, lr = 1e-13
I0929 10:39:21.761581  4800 solver.cpp:242] Iteration 1200, loss = 49971.9
I0929 10:39:21.761611  4800 solver.cpp:258]     Train net output #0: loss = 49971.9 (* 1 = 49971.9 loss)
I0929 10:39:21.761615  4800 solver.cpp:571] Iteration 1200, lr = 1e-13
I0929 10:39:44.492492  4800 solver.cpp:242] Iteration 1250, loss = 25675.8
I0929 10:39:44.492522  4800 solver.cpp:258]     Train net output #0: loss = 25675.8 (* 1 = 25675.8 loss)
I0929 10:39:44.492527  4800 solver.cpp:571] Iteration 1250, lr = 1e-13
I0929 10:40:07.287600  4800 solver.cpp:242] Iteration 1300, loss = 26559.2
I0929 10:40:07.287631  4800 solver.cpp:258]     Train net output #0: loss = 26559.2 (* 1 = 26559.2 loss)
I0929 10:40:07.287636  4800 solver.cpp:571] Iteration 1300, lr = 1e-13
I0929 10:40:21.817070  4800 solver.cpp:346] Iteration 1333, Testing net (#0)
F0929 10:40:21.817103  4800 syncedmem.hpp:30] Check failed: error == cudaSuccess (11 vs. 0)  invalid argument
*** Check failure stack trace: ***


I'm using GTX 1050 4G for training with cuda-8.0. I had resize training image to 100px. Can you help me?

**Comment by RoseLii** [comment]
I have the same problem,please help me.
Cannot copy param 0 weights from layer 'conv4_3_norm_mbox_conf'; shape mismatch.  Source param shape is 84 512 3 3 (387072); target param shape is 176 512 3 3 (811008). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer. @caffecoffee @qincao1994

**Comment by Ram-Godavarthi** [comment]
@RoseLii Did you happen to solve the problem?
